pallets/statistics/src/lib.rs

```diff
@@ -14,25 +14,167 @@
+    type MaxTransferManagersPerAsset: Get<u32>;
+    /// Weights for extrinsics
+    type WeightInfo: WeightInfo;
+}
+
+/// Weight info for extrinsics
+pub trait WeightInfo {
+    fn add_transfer_manager() -> Weight;
+    fn remove_transfer_manager() -> Weight;
+    fn add_exempted_entities(i: u32) -> Weight;
+    fn remove_exempted_entities(i: u32) -> Weight;
+}
 
 decl_storage! {
     trait Store for Module<T: Trait> as statistics {
-        /// Number of investor per asset.
-        pub InvestorCountPerAsset get(fn investor_count_per_asset): map hasher(blake2_128_concat) Ticker => Counter ;
+        /// Transfer managers currently enabled for an Asset.
+        pub ActiveTransferManagers get(fn transfer_managers): map hasher(blake2_128_concat) Ticker => Vec<TransferManager>;
+        /// Number of current investors in an asset.
+        pub InvestorCountPerAsset get(fn investor_count): map hasher(blake2_128_concat) Ticker => Counter;
+        /// Entities exempt from transfer managers. Exemptions requirements are based on TMS.
+        /// TMs may require just the sender, just the receiver, both or either to be exempted.
+        /// CTM requires sender to be exempted while PTM requires receiver to be exempted.
+        pub ExemptEntities get(fn entity_exempt):
+            double_map
+                hasher(blake2_128_concat) (Ticker, TransferManager),
+                hasher(blake2_128_concat) ScopeId
+            =>
+                bool;
     }
 }
 
 decl_module! {
     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        /// initialize the default event for this module
+        fn deposit_event() = default;
+
+        /// Adds a new transfer manager.
+        ///
+        /// # Arguments
+        /// * `origin` It contains the secondary key of the caller (i.e who signed the transaction to execute this function).
+        /// * `ticker` ticker for which the transfer managers are being updated.
+        /// * `new_transfer_manager` Transfer manager being added.
+        ///
+        /// # Errors
+        /// * `Unauthorized` if `origin` is not the owner of the ticker.
+        /// * `DuplicateTransferManager` if `new_transfer_manager` is already enabled for the ticker.
+        /// * `TransferManagersLimitReached` if the `ticker` already has max TMs attached
+        ///
+        #[weight = <T as Trait>::WeightInfo::add_transfer_manager()]
+        pub fn add_transfer_manager(origin, ticker: Ticker, new_transfer_manager: TransferManager) {
+            let did = T::Asset::ensure_perms_owner_asset(origin, &ticker)?;
+            ActiveTransferManagers::try_mutate(&ticker, |transfer_managers| {
+                ensure!((transfer_managers.len() as u32) < T::MaxTransferManagersPerAsset::get(), Error::<T>::TransferManagersLimitReached);
+                ensure!(!transfer_managers.contains(&new_transfer_manager), Error::<T>::DuplicateTransferManager);
```

Comment:
Should we prevent duplicates at the "base" enum level - i.e. only a single `CountTransferManager` allowed, regardless of its `Counter` parameter?

Reply:
It's not a bug, it's a feature. 🤭